infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1195,6 +1186,7 @@ private void closeSegmentHandlers(long endSegment) {
                         sh.getSegment(), sh.getRefCount()
                 );
             }
+            channelsToSync.remove(sh.getWriteChannel());
```

Comment:
Why is this needed? `sh.close();` will flush internal channels.

Reply:
This is to ensure that next time when sync() is called by batchProcessor, there are no closed channels in the channelsToSync map.